### PR TITLE
Clean up remaining DISABLE_LEVEL_CHECKS ifndefs

### PR DESCRIPTION
**Pull request by MSpiller7**
This replaces the remaining `DISABLE_LEVEL_CHECKS` `ifndefs` with correct `ENABLE_VANILLA_LEVEL_SPECIFIC_CHECKS` `ifdefs`. One of the vanilla checks in `src/game/level_update.c` appears to do the same thing as another vanilla check in the same file (namely, make double super extra sure the slide music doesn't get canceled going through a warp in BOB); could we look into whether we could remove one of those?

### CODE REVIEW

**Review by thecozies — APPROVED** [review]
AYYYYY

**Review by gheskett — APPROVED** [review]
Wtf I didn't even know this was a thing

File: src/game/level_update.c [hunk 1]
```diff
@@ -552,7 +552,7 @@ s16 music_unchanged_through_warp(s16 arg) {
     s16 unchanged = TRUE;
     s16 currBgMusic;
```
Comment (gheskett) [inline src/game/level_update.c:553]: `currBgMusic` and `levelNum` are unused and will appropriately throw warnings here, can you move these declarations inside of the ifdef?

**Review by gheskett — CHANGES_REQUESTED** [review]

File: src/game/level_update.c (hunk 1 above)
Comment (MSpiller7) [reply src/game/level_update.c]: Hopefully fixed this in my latest commit, thank you for the heads up!

**Review by gheskett — APPROVED** [review]

**Comment by MSpiller7** [comment]
oops lol

**Review by gheskett — APPROVED** [review]

**Pull request merged**